node-launchpad/src/components/home.rs

```diff
@@ -328,72 +377,106 @@ impl Component for Home {
-        let node_registry = NodeRegistry::load(&get_node_registry_path()?)?;
-        self.running_nodes = node_registry
-            .nodes
-            .into_iter()
-            .filter(|node| node.status != ServiceStatus::Removed)
-            .collect();
-        info!(
-            "Loaded node registry. Runnign nodes: {:?}",
-            self.running_nodes.len()
-        );
-
-        if !self.running_nodes.is_empty() && self.node_table_state.selected().is_none() {
-            self.node_table_state.select(Some(0));
+fn stop_nodes(services: Vec<String>, action_sender: UnboundedSender<Action>) {
+    tokio::task::spawn_local(async move {
+        if let Err(err) = sn_node_manager::cmd::node::stop(
+            vec![],
+            services,
+            sn_node_manager::VerbosityLevel::Minimal,
+        )
+        .await
+        {
+            error!("Error while stopping services {err:?}");
+        } else {
+            info!("Successfully stopped services");
+        }
+        if let Err(err) = action_sender.send(Action::HomeActions(
+            HomeActions::ServiceManagerOperationCompleted,
+        )) {
+            error!("Error while sending action: {err:?}");
         }
+    });
+}
 
-        Ok(())
-    }
+fn start_nodes(services: Vec<String>, action_sender: UnboundedSender<Action>) {
+    tokio::task::spawn_local(async move {
+        // I think using 1 thread is causing us to block on the below start function and not really
+        // having a chance to set lock_registry = true and draw from that state. Since the update is slow,
+        // the gui looks laggy. Adding a sleep basically puts this to sleep while drawing with the new state.
+        tokio::time::sleep(Duration::from_millis(100)).await;
+        if let Err(err) = sn_node_manager::cmd::node::start(
+            NODE_START_INTERVAL as u64,
+            vec![],
+            services,
+            sn_node_manager::VerbosityLevel::Minimal,
+        )
+        .await
+        {
+            error!("Error while starting services {err:?}");
+        } else {
+            info!("Successfully started services");
+        }
+        if let Err(err) = action_sender.send(Action::HomeActions(
+            HomeActions::ServiceManagerOperationCompleted,
+        )) {
+            error!("Error while sending action: {err:?}");
+        }
+    });
+}
 
-    fn select_next_table_item(&mut self) {
-        let i = match self.node_table_state.selected() {
-            Some(i) => {
-                if i >= self.running_nodes.len() - 1 {
-                    0
-                } else {
-                    i + 1
-                }
+fn add_and_start_nodes(
+    count: usize,
+    mut nodes_to_start: Vec<String>,
+    action_sender: UnboundedSender<Action>,
+) {
+    let peers = PeersArgs::default(); // will fetch from network contacts
+
+    tokio::task::spawn_local(async move {
+        let result = sn_node_manager::cmd::node::add(
+            Some(count as u16),
```

Comment:
should this not be `nodes_to_start` ?

Reply:
`nodes_to_start` denotes the inactive nodes that we are supposed to restart basically.